### PR TITLE
Withdraw funds issue

### PR DESCRIPTION
**Pull request by tommyrharper**

### CODE REVIEW

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol [hunk 1]
```diff
@@ -238,15 +240,58 @@ contract PerpsAccountModule is IPerpsAccountModule {
                 amount
             );
         } else {
```
Comment (JaredBorders) [inline markets/perps-market/contracts/modules/PerpsAccountModule.sol:242]: If quanto markets only support margin denominated in the quanto asset, we need to remove all logic not related to quanto asset-based collateral outside of reverts, checks, etc

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol [hunk 2]
```diff
@@ -238,15 +240,58 @@ contract PerpsAccountModule is IPerpsAccountModule {
...
+            ISpotMarketSystem spotMarket = perpsMarketFactory.spotMarket;
             ITokenModule synth = ITokenModule(
-                perpsMarketFactory.spotMarket.getSynth(synthMarketId)
+                spotMarket.getSynth(synthMarketId)
```
Comment (JaredBorders) [inline markets/perps-market/contracts/modules/PerpsAccountModule.sol:245]: `synthMarketId` should only be the quanto asset synth id. there needs to be a check if so

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol [hunk 3]
```diff
@@ -238,15 +240,58 @@ contract PerpsAccountModule is IPerpsAccountModule {
...
+
+            uint256 collateralAmount = perpsMarketFactory.synthetix.getMarketCollateralAmount(
                 perpsMarketId,
-                address(synth),
-                amount
+                address(synth)
             );
+
+            // can withdraw all the payout from the markets collateral directly
+            if (collateralAmount >= amount) {
```
Comment (JaredBorders) [inline markets/perps-market/contracts/modules/PerpsAccountModule.sol:254]: is this safe? can the value returned from `perpsMarketFactory.synthetix.getMarketCollateralAmount` be safely withdrawn 100% of the time?

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol [hunk 4]
```diff
@@ -238,15 +240,58 @@ contract PerpsAccountModule is IPerpsAccountModule {
...
+                // withdrawing from a synth market
```
Comment (JaredBorders) [inline markets/perps-market/contracts/modules/PerpsAccountModule.sol:255]: this is withdrawing from a perps market, right?

**Review by JaredBorders — CHANGES_REQUESTED** [review]

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol (hunk 2 above)
Comment (tommyrharper) [reply markets/perps-market/contracts/modules/PerpsAccountModule.sol:245]: I am thinking we should keep it flexible.

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol (hunk 4 above)
Comment (tommyrharper) [reply markets/perps-market/contracts/modules/PerpsAccountModule.sol]: This is a bad comment - fixed!

File: markets/perps-market/contracts/modules/PerpsAccountModule.sol (hunk 2 above)
Comment (tommyrharper) [reply markets/perps-market/contracts/modules/PerpsAccountModule.sol:245]: Yeah I think this is fine, based on the fact that we are not removing multi-collateral for the moment.

**Review by JaredBorders — APPROVED** [review]

**Pull request merged**